File: MoSmartPark/MoSmartPark.Services/Database/Reservation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MoSmartPark.Services.Database
{
    public class Reservation
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int CarId { get; set; }

        [Required]
        public int ParkingSpotId { get; set; }

        [Required]
        public int ReservationTypeId { get; set; }

        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal FinalPrice { get; set; }

        public string? QrCodeData { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public Car Car { get; set; } = null!;
        public ParkingSpot ParkingSpot { get; set; } = null!;
        public ReservationType ReservationType { get; set; } = null!;
    }
}

File: MoSmartPark/UI/mosmartpark_desktop/lib/model/reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  final int id;
  final int carId;
  final int parkingSpotId;
  final int reservationTypeId;
  final DateTime? startDate;
  final DateTime? endDate;
  final double finalPrice;
  final DateTime createdAt;
  
  // Navigation property details
  final String? carModel;
  final String? carBrandName;
  final String? carLicensePlate;
  final String? carColorName;
  final String? carColorHexCode;
  final String? parkingSpotNumber;
  final String? reservationTypeName;
  final String? userFullName;
  final String? userPicture; // Will be base64 string from backend byte[]

  Reservation({
    this.id = 0,
    this.carId = 0,
    this.parkingSpotId = 0,
    this.reservationTypeId = 0,
    this.startDate,
    this.endDate,
    this.finalPrice = 0.0,
    required this.createdAt,
    this.carModel,
    this.carBrandName,
    this.carLicensePlate,
    this.carColorName,
    this.carColorHexCode,
    this.parkingSpotNumber,
    this.reservationTypeName,
    this.userFullName,
    this.userPicture,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) => _$ReservationFromJson(json);
  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}

